core/platform/bulk_email/mailchimp_bulk_email_services.py

```diff
@@ -77,16 +77,16 @@ def _get_mailchimp_class() -> Optional[mailchimp3.MailChimp]:
 
 
 def _create_user_in_mailchimp_db(
-    client: mailchimp3.MailChimp, user_email: str
+    client: mailchimp3.MailChimp,
+    subscribed_mailchimp_data: Dict[str, Sequence[str]]
 ) -> bool:
     """Creates a new user in the mailchimp database and handles the case where
     the user was permanently deleted from the database.
 
     Args:
         client: mailchimp3.MailChimp. A mailchimp instance with the API key and
             username initialized.
-        user_email: str. Email ID of the user. Email is used to uniquely
-            identify the user in the mailchimp DB.
+        subscribed_mailchimp_data: dict. Post body with required fields.
```

Comment:
Could you please explain the internal structure of this more?

If there are fields that need to be provided each time, split those out in the arguments and compose the dict internally here -- better to be explicit within our codebase.

Reply:
This is a private function that should only be called from within this file and the function that calls this has the relevant fields separate in its arguments. So, isn't that fine? The dict is constructed within this script.

Reply:
Done, expanded internal structure in the comment, but above question still stands.

Reply:
It mitigates it slightly but the general recommendation still stands -- we try to keep the function interfaces as explicit as possible for determinism and for ease of developers seeing what the boundary is.